### PR TITLE
Fix nested data dir variable names

### LINKED ISSUE #28: undefined method `data' for #<Frontman::App:0x0000562ba8230c60>
Using `data` within any view template results in the following error.

Just following the instructions on https://github.com/algolia/frontman/wiki/Data-files

### PR DESCRIPTION
**Pull request by westonganger**
Fixes #28 

We also need to document that the variable names are set by the data dir name.

### CODE REVIEW

**Comment by westonganger** [comment]
Is this `define_singleton_method` a good pattern? Does it leave possibility for name clashes with existing methods. Or consider the following example?

```ruby
register_data_dirs([
  'site/data',
  'other/data',
])
```

Another comment, I would almost prefer that all data is nested under the single `data` method but maybe thats not flexible enough.

**Comment by DevinCodes** [comment]
I think in the example you're providing, we would want to throw an error. IMO, data files should be:

* placed in a single directory, that you then register. (in your example, there's two ways to address this: register the `site` and `other` directory and access everything with `site.data.x` or `other.data.x`, or have a `data` directory with `other` and `site` directories in it. You would only register the `data` directory). 
* Or, have a unique name so that there's no chance of collision.

Either way, the names of the data directory should not collide, and if it's the case I think we should raise an error so that the user can rename or restructure the data files. 

What are your thoughts on this?

**Comment by westonganger** [comment]
Yes I think the error based approach for conflicts could work.

Heres another idea which makes the variable definition a lot more explicit and understandable.

```ruby
register_data_dirs([
  site_data: 'site/data',
  other_data: 'other/data',
])

# Then access via
site_data.title
# OR
data(:site_data).title ### this format wont pollute the global namespace.
```

**Comment by DevinCodes** [comment]
I like the API you're suggesting, it is indeed clear and a lot more explicit.

However, it would introduce a breaking change for what I believe to be an optimization for a situation that shouldn't arise. In theory a site could have multiple data directories, in practice I don't think this will happen too often (e.g. Middleman only supports one data directory, which doesn't seem to generate too much complaints).

For now, I believe we should go with the error-based approach for conflicts. However, let's keep an eye on how people use this: if it turns out that the current working is quite problematic for our users, I have no issue to refactor this into something more like you proposed.

I would love to get @sarahdayan 's opinion on the topic, feel free to join the discussion!

**Comment by westonganger** [comment]
I agree the breaking change is not needed as multiple folders are an edge case. However I would like to be able to support the renaming approach in a backwards compatible way. 

For example if my data folder is named `yaml` but I want to be able to use the `data` in my templates.

```ruby
### Legacy Syntax
register_data_dirs([
  'app/yaml'
])

yaml.site.title

### Alternate Syntax
register_data_dirs({
  data: 'app/yaml',
})

data.site.title
```

**Comment by DevinCodes** [comment]
If we can keep backwards compatibility, I'm all in favor of adding this! It's an awesome addition, thank you @westonganger  😄

**Comment by westonganger** [comment]
Ok I have now updated this PR.

File: lib/frontman/app.rb [hunk 1]
```diff
@@ -94,11 +94,21 @@ def add_to_manifest(key, value)
...
-        define_singleton_method dir.to_sym do
-          @data_dirs[dir] ||= DataStore.new(File.join(Dir.pwd, dir))
+      if dirs.is_a?(Array)
+        array = dirs.dup
+
+        dirs = {}
+
+        array.each do |dir|
+          dirs[dir.split("/").last] = dir
+        end
```
Comment (DevinCodes) [inline lib/frontman/app.rb:106]: We could probably refactor this to a one-liner, something like this:

```ruby
dirs = dirs.map { |dir| [dir.split("/").last, dir] }.to_h
```

WDYT?

**Review by DevinCodes — CHANGES_REQUESTED** [review]

**Review by DevinCodes — APPROVED** [review]

**Comment by DevinCodes** [comment]
Updated the docs to reflect this change: https://github.com/algolia/frontman/wiki/Data-files

**Comment by DevinCodes** [comment]
We just released this in version [`0.0.4`](https://rubygems.org/gems/frontman-ssg/versions/0.0.4), thank you for the contribution! 🎉

**Pull request merged**